pkg/reconciler/instances/istio/action.go

```diff
@@ -1,37 +1,104 @@
 package istio
 
 import (
-	"fmt"
+	"bytes"
 	"github.com/kyma-incubator/reconciler/pkg/reconciler"
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/chart"
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/file"
 	"github.com/kyma-incubator/reconciler/pkg/reconciler/service"
 	"github.com/pkg/errors"
+	"io"
+	"os"
 	"os/exec"
+	"strings"
 )
 
 const (
-	istioctl1_10_2 = "/bin/istioctl-1.10.2"
+	istioctl1_11_1      = "/bin/istioctl-1.11.1" // change path to the local `istioctl` if debugging locally
+	yaml_delimiter      = "---"
+	istio_operator_kind = "kind: IstioOperator"
 )
 
 type ReconcileAction struct {
 }
 
 func (a *ReconcileAction) Run(version, profile string, config []reconciler.Configuration, context *service.ActionContext) error {
-	ws, err := context.WorkspaceFactory.Get(version)
+	var overrides = make(map[string]interface{}, len(config))
+
+	for _, configEntry := range config {
+		overrides[configEntry.Key] = configEntry.Value
+	}
+
+	comp := chart.NewComponent("istio-configuration", "istio-system", overrides)
+	componentSet := chart.NewComponentSet(context.KubeClient.Kubeconfig(), version, profile, []*chart.Component{comp})
+	manifests, err := context.ChartProvider.Manifests(componentSet, false, &chart.Options{})
+	if err != nil {
+		return err
+	}
+
+	manifestsCount := len(manifests)
+	if manifestsCount != 1 {
+		return errors.Errorf("One manifest expected but got %d", manifestsCount)
+	}
+
+	finalManifest := manifests[0].Manifest
+	istioOperator := extractIstioOperatorContextFrom(finalManifest)
+	istioOperatorPath, istioOperatorCf, err := file.CreateTempFileWith(istioOperator)
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		cleanupErr := istioOperatorCf()
+		if cleanupErr != nil {
+			context.Logger.Error(cleanupErr)
+		}
+	}()
+
+	kubeconfigPath, kubeconfigCf, err := file.CreateTempFileWith(context.KubeClient.Kubeconfig())
 	if err != nil {
-		return errors.Wrap(err,
-			fmt.Sprintf("Failed to retrieve Kyma workspace '%s' in ISTIO reconcile-action", version))
+		return err
 	}
-	fmt.Printf("Kyma sources are located here: %s\n", ws.WorkspaceDir)
 
-	return istioctl(version)
+	defer func() {
+		cleanupErr := kubeconfigCf()
```

Comment:
MAybe we can make the name a bit more expressive like `deleteTmpFile` or similar... Cf stands for cleanup-Fct?